megamek/src/megamek/client/ui/swing/lobby/ChatLounge.java

```diff
@@ -0,0 +1,3628 @@
+            }
+            return null;
+        }
+    }
+
+    Map<String, ImageIcon> mapIcons = new HashMap<String, ImageIcon>();
+    
+    /** A renderer for the list of available boards. */
+    public class BoardNameRenderer extends DefaultListCellRenderer  {
+        private static final long serialVersionUID = -3218595828938299222L;
+        
+        private float oldGUIScale = GUIPreferences.getInstance().getGUIScale();
+        private Image image;
+        private ImageIcon icon;
+        
+        @Override
+        public Component getListCellRendererComponent(JList<?> list, Object value, 
+                int index, boolean isSelected, boolean cellHasFocus) {
+
+            String board = (String)value;
+            // For generated boards, add the size to have different images for different sizes
+            if (board.startsWith(MapSettings.BOARD_GENERATED)) {
+                board += mapSettings.getBoardSize().toString();
+            }
+            
+            // If the gui scaling has changed, clear out all images, triggering a reload
+            float currentGUIScale = GUIPreferences.getInstance().getGUIScale();
+            if (currentGUIScale != oldGUIScale) {
+                oldGUIScale = currentGUIScale;
+                mapIcons.clear();
+                synchronized (baseImages) {
+                    baseImages.clear();
+                }
+            }
+            
+            // If an icon is present for the current board, use it
+            icon = mapIcons.get(board);
+            if (icon != null) {
+                setIcon(icon);
+            } else {
+                // The icon is not present, see if there's a base image
+                synchronized (baseImages) {
+                    image = baseImages.get(board);
+                }
+                if (image == null) {
+                    // There's no base image: trigger loading it and, for now, return the base list's panel
+                    // The [GENERATED] entry will always land here as well
+                    loader.add(board);
+                    setToolTipText(null);
+                    return super.getListCellRendererComponent(list, new File(board).getName(), index, isSelected, cellHasFocus);
+                } else {
+                    // There is a base image: make it into an icon, store it and use it
+                    if (!lisBoardsAvailable.isEnabled()) {
+                        ImageFilter filter = new GrayFilter(true, 50);  
+                        ImageProducer producer = new FilteredImageSource(image.getSource(), filter);  
+                        image = Toolkit.getDefaultToolkit().createImage(producer); 
+                    }
+                    icon = new ImageIcon(image);
+                    
+                    mapIcons.put(board, icon);
+                    setIcon(icon);
+                }
+            }
+            
+            // Found or created an icon; finish the panel
+            setText("");
```

Comment:
When an icon isn't there, the super renderer result is returned a few lines above. This will set the text, so it has to be reset when an icon is found and used (otherwise, the text of a first list entry will stay present for all subsequently rendered list entries, sorry if that overexplains).